docs/system_overview/genquery.md

```diff
@@ -553,3 +551,283 @@ Find all data objects with replicas on resource `replResc` which are not marked
+```
+
+GenQuery2 will notice the use of double single quotes and collapse them to one single quote before sending to the database. In the case of the example, that means `that''s` will be passed to the database as `that's`.
+
+### Showing Duplicate Entries
+
+By default, GenQuery2 removes duplicate entries from the resultset.
+
+If you need to view duplicate entries, add `no distinct` after the `select` keyword. For example:
+
+```sh
+iquery "select no distinct COLL_NAME, DATA_NAME"
+```
+
+### Group-By Clause
+
+GenQuery2 provides support for the GROUP-BY clause. The important thing to remember about GROUP-BY is that if the PROJECTION clause contains aggregation operators or SQL functions, the GROUP-BY clause may also require those same aggregation operators and/or SQL functions.
+
+Here's an example that calculates the number of data objects in each collection.
+
+```sh
+iquery "select COLL_NAME, count(DATA_NAME) group by COLL_NAME"
+```
+
+### Offsets and Limiting the size of a resultset
+
+By default, GenQuery2 will return a max of 256 rows if no limit is applied to the query.
+
+You can change the size of the resultset by specifying a `LIMIT` or `FETCH FIRST N ROWS ONLY` clause. Both achieve the same outcome.
+
+For example:
+
+```sh
+iquery "select COLL_NAME, DATA_NAME limit 1000"
+```
+
+!!! Note
+    Keep in mind that GenQuery2 does NOT provide built-in pagination. It will always return the resultset in its entirety!
+
+You can also apply an offset by using the `OFFSET` keyword.
+
+```sh
+iquery "select COLL_NAME, DATA_NAME order by COLL_NAME, DATA_NAME offset 2000 limit 1000"
+```
+
+Notice we're now also sorting the results. Without the ORDER-BY clause, the offset and limit add little value.
+
+These features are great for relatively small datasets. However, as your datasets grow in size, you may want to build more sophisticated pagination solution.
+
+### Resultsets and Pagination
+
+Unlike GenQuery1, GenQuery2 does not provide built-in support for pagination. The resultset of a query will always be returned in its entirety. That means users fetching large amounts of data need to be careful of exhausting memory resources on the iRODS servers. Thankfully, GenQuery2 helps with this by defaulting to 256 rows if `LIMIT` is not defined in the query.
+
+!!! Note
+    `LIMIT` is provided as a convenience. It can be used in place of `FETCH FIRST N ROWS ONLY`, which is what's defined as part of the SQL standard.
+
+While GenQuery2 doesn't provide built-in support for pagination, it does provide enough features for users to implement pagination efficiently. This is left as an exercise for the user.
+
+### Other Notable Features
+
+GenQuery2 supports the following additional features:
+
+- Logical AND, OR, and NOT
+- Grouping via parentheses
+- SQL CAST
+- Full range of relational operators: =, !=, <, <=, >, >=, LIKE, BETWEEN, IS [NOT] NULL
+- SQL keywords are case-insensitive
+- Federation
+
+GenQuery2 shares the same syntax as standard SQL.
```

Comment:
Consider moving this to the top of the GenQuery2 section.

Also, consider making examples for parents, etc.

Reply:
Done.